pallets/file-system/src/benchmarking.rs

```diff
@@ -0,0 +1,165 @@
+        + shp_traits::ReadProvidersInterface<ProviderId = <T as frame_system::Config>::Hash>,
+    // Ensure the ValuePropId from our Providers trait matches that from pallet_storage_providers:
+    <T as crate::Config>::Providers: shp_traits::MutateBucketsInterface<ValuePropId = <T as pallet_storage_providers::Config>::ValuePropId>,
+)]
+mod benchmarks {
+    use super::*;
+    use frame_support::{
+        assert_ok,
+        traits::{fungible::Mutate, Get},
+        BoundedVec,
+    };
+    use frame_system::RawOrigin;
+    use pallet_storage_providers::types::ValueProposition;
+    use shp_traits::ReadBucketsInterface;
+    use sp_core::Hasher;
+    use sp_runtime::traits::Hash;
+    use sp_std::vec;
+
+    #[benchmark]
+    fn create_bucket() -> Result<(), BenchmarkError> {
+        let user: T::AccountId = account("Alice", 0, 0);
+        let signed_origin = RawOrigin::Signed(user.clone());
+        mint_into_account::<T>(user.clone(), 1_000_000_000_000_000)?;
+
+        let name: BucketNameFor<T> = vec![1; BucketNameLimitFor::<T>::get().try_into().unwrap()]
+            .try_into()
+            .unwrap();
+
+        // Register MSP with value proposition
+        let msp: T::AccountId = account("MSP", 0, 0);
+        mint_into_account::<T>(msp.clone(), 1_000_000_000_000_000)?;
+        let (msp_id, value_prop_id) = add_msp_to_provider_storage::<T>(&msp);
+
+        #[extrinsic_call]
+        _(
+            signed_origin.clone(),
+            Some(msp_id),
+            name,
+            true,
+            Some(value_prop_id),
+        );
+
+        Ok(())
+    }
+
+    #[benchmark]
+    fn issue_storage_request() -> Result<(), BenchmarkError> {
+        let user: T::AccountId = account("Alice", 0, 0);
+        let signed_origin = RawOrigin::Signed(user.clone());
+        mint_into_account::<T>(user.clone(), 1_000_000_000_000_000)?;
+
+        let name: BucketNameFor<T> = vec![1; BucketNameLimitFor::<T>::get().try_into().unwrap()]
+            .try_into()
+            .unwrap();
+        let bucket_id = <<T as crate::Config>::Providers as ReadBucketsInterface>::derive_bucket_id(
+            &user,
+            name.clone(),
+        );
+        let location = vec![1; MaxFilePathSize::<T>::get().try_into().unwrap()]
+            .try_into()
+            .unwrap();
+        let fingerprint = <<T as frame_system::Config>::Hashing as Hasher>::hash(b"tes");
+        let size: StorageData<T> = 100;
+        let peer_id = BoundedVec::try_from(vec![1]).unwrap();
+        let peer_ids: PeerIds<T> = BoundedVec::try_from(vec![peer_id]).unwrap();
+
+        Pallet::<T>::create_bucket(signed_origin.clone().into(), None, name, true, None)?;
+
+        #[extrinsic_call]
+        _(
+            signed_origin,
+            bucket_id,
+            location,
+            fingerprint,
+            size,
+            None,
+            peer_ids,
+        );
```

Comment:
Worst-case scenario I believe is:
- With a MSP (since there are extra checks that the pallet has to do)
- With a full list of peer IDs (since the logic clones the list)